new.go

```diff
@@ -277,23 +297,33 @@ func newBuilder(ctx context.Context, store storage.Store, options BuilderOptions
 			name = imageNamePrefix(image) + "-" + name
 		}
 	}
+	var container *storage.Container
+	suffix := 0
+	tmpName := name
+	if options.Container == "" {
+		containers, err := store.Containers()
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to check for container names")
+		}
+		tmpName, suffix = findUnusedContainer(tmpName, containers)
+	}
 
-	coptions := storage.ContainerOptions{}
-	coptions.IDMappingOptions = newContainerIDMappingOptions(options.IDMappingOptions)
-
-	container, err := store.CreateContainer("", []string{name}, imageID, "", "", &coptions)
-	suffix := 1
-	for err != nil && errors.Cause(err) == storage.ErrDuplicateName && options.Container == "" {
-		suffix++
-		tmpName := fmt.Sprintf("%s-%d", name, suffix)
-		if container, err = store.CreateContainer("", []string{tmpName}, imageID, "", "", &coptions); err == nil {
+	for true {
+		coptions := storage.ContainerOptions{
+			LabelOpts:        options.CommonBuildOpts.LabelOpts,
+			IDMappingOptions: newContainerIDMappingOptions(options.IDMappingOptions),
+		}
+		container, err = store.CreateContainer("", []string{tmpName}, imageID, "", "", &coptions)
```

Comment:
Name conflicts would still appear here in parallel builds. I am not really not found of the incremental suffixes. They look much nicer but it's really expensive when hitting this condition.

Reply:
Well the conflict is only going to happen for containers that are created since the buildah command is executed.  Only Buildah uses this algorythm.  A conflict would only happen if a second buildah rand creating a container since the first container read the container store.  Unlike the original code this buildah code will only try on a known `good` name based on the list of containers that were created at start.